launch.go

```diff
@@ -270,19 +270,90 @@ func createMetaSpace(metaSpace string) error {
 }
 
 func writeMetafile(metaSpace, metaFile, metaLog string, mergedMeta map[string]interface{}) error {
-	metaByte := []byte("")
-	log.Println("Marshalling Merged Meta JSON")
-	metaByte, err := marshal(mergedMeta)
+	var metaPath = metaSpace + "/" + metaFile
+
+	_, err := os.Stat(metaPath)
+
+	// If file doesn't already exist, write to new file
+	if os.IsNotExist(err) {
+		metaByte := []byte("")
+		log.Println("Marshalling Merged Meta JSON in writeMetafile")
+		metaByte, err := marshal(mergedMeta)
+
+		if err != nil {
+			return fmt.Errorf("Parsing Meta JSON: %v", err)
+		}
+
+		err = writeFile(metaPath, metaByte, 0666)
+		if err != nil {
+			return fmt.Errorf("Writing Parent %v Meta JSON: %v", metaLog, err)
+		}
+	} else {
+		// If file already exists, merge meta in file
+		// Get meta from file
+		jsonFile, err := os.Open(metaPath)
+		if err != nil {
+			return fmt.Errorf("Opening external Meta JSON %s: %v", metaPath, err)
+		}
+		defer jsonFile.Close()
+
+		oldMetaByte, _ := ioutil.ReadAll(jsonFile)
+
+		var oldMeta map[string]interface{}
+		json.Unmarshal([]byte(oldMetaByte), &oldMeta)
+
+		// merge with new meta
+		mergedExternalMeta := deepMergeJSON(mergedMeta, oldMeta)
+
+		metaByte := []byte("")
+		log.Println("Marshalling Merged Meta JSON in writeMetafile")
+		metaByte, err = marshal(mergedExternalMeta)
+
+		if err != nil {
+			return fmt.Errorf("Parsing Meta JSON: %v", err)
+		}
+
+		// save meta
+		err = writeFile(metaPath, metaByte, 0666)
+		if err != nil {
+			return fmt.Errorf("Writing Parent %v Meta JSON: %v", metaLog, err)
+		}
+	}
+
+	return nil
+}
 
+// SetExternalMeta checks if parent build is external and sets meta in external file accordingly
```

Comment:
is writing to external meta file enough? meta cli looks at `meta key` first https://github.com/screwdriver-cd/meta-cli/blob/9cfbdc987b7c55e6a03559f0d793d1f321124fa4/meta.go#L77

So even if we update the external file, won't `meta key` still have old value ? I think updating the `meta key` is also needed